sdk/python/tests/integration/feature_repos/repo_configuration.py

```diff
@@ -0,0 +1,242 @@
+import tempfile
+import uuid
+from contextlib import contextmanager
+from dataclasses import dataclass, field, replace
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+
+from feast import FeatureStore, FeatureView, RepoConfig, driver_test_data, importer
+from feast.data_source import DataSource
+from tests.integration.feature_repos.universal.data_source_creator import (
+    DataSourceCreator,
+)
+from tests.integration.feature_repos.universal.feature_views import (
+    create_customer_daily_profile_feature_view,
+    create_driver_hourly_stats_feature_view,
+)
+
+
+@dataclass(frozen=True, repr=True)
+class IntegrationTestRepoConfig:
+    """
+    This class should hold all possible parameters that may need to be varied by individual tests.
+    """
+
+    provider: str = "local"
+    online_store: Union[str, Dict] = "sqlite"
+
+    offline_store_creator: str = "tests.integration.feature_repos.universal.data_sources.file.FileDataSourceCreator"
```

Comment:
Apologies if this has been mentioned before, but can you remind me why we use a string reference to the class instead of a code reference?

Reply:
Is there any downside to having that code dependency?

Reply:
Oops missed this. For tests, presumably not. However I can imagine we'll want this to be extensible to support the test suite for when users bring their own offline stores with a custom DataSourceCreator, so it may be needed then. Happy to do either.

Reply:
My personal preference is to YAGNI this baby until we need it, just because I feel like we might run into bugs with these strings.

Reply:
Aight sg